### PR TITLE
Support dynamic end for tf.strided_slice conversion

### PR DESCRIPTION
**Pull request by jiafatom**

### CODE REVIEW

File: keras2onnx/_builtin.py [hunk 1]
```diff
@@ -1875,8 +1889,29 @@ def convert_tf_strided_slice(scope, operator, container):
...
+                                operator.inputs[0].full_name + '_shape')
+    data_shape_mul = oopb.add_node('Mul',
+                                   [data_shape,
+                                    ('_start', oopb.int64, np.array(end_mask_array, dtype=np.int64))],
+                                   operator.inputs[0].full_name + '_shape_mul')
+    end_mask_array_neg = 1 - np.array(end_mask_array, dtype=np.int64)
+    end_cast_0 = oopb.add_node('Cast',
+                               node.inputs[2].name,
+                               node.inputs[2].name + '_end_cast_0', to=7)
+    end_cast_0_mul = oopb.add_node('Mul',
```
Comment (wenbingl) [inline keras2onnx/_builtin.py:1903]: should it be 'apply_mul' and apply_cast?

Comment (jiafatom) [reply keras2onnx/_builtin.py]: done

File: keras2onnx/_builtin.py [hunk 2]
```diff
@@ -1875,8 +1889,26 @@ def convert_tf_strided_slice(scope, operator, container):
     else:
         new_axis_unsqueeze = operator.inputs[0].full_name
 
+    data_shape = oopb.add_node('Shape',
+                                operator.inputs[0].full_name,
+                                operator.inputs[0].full_name + '_shape')
```
Comment (wenbingl) [inline keras2onnx/_builtin.py:1894]: always need a op_version for add_node method.

Comment (jiafatom) [reply keras2onnx/_builtin.py:1894]: done

**Pull request merged**